cumulusci/core/tests/test_config.py

```diff
@@ -210,20 +248,328 @@ def get_service(self, name):
+            os.chdir(d)
+            self.assertIsNone(config.repo_url)
+            os.chdir(cwd)
+
+    def test_repo_url_from_git(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        self.assertIn('SalesforceFoundation/CumulusCI', config.repo_url)
+
+    def test_repo_owner_from_repo_info(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        config._repo_info = {
+            'owner': 'SalesforceFoundation',
+        }
+        self.assertEqual(
+            'SalesforceFoundation', config.repo_owner)
+
+    def test_repo_owner_no_repo_root(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        with temporary_dir() as d:
+            cwd = os.getcwd()
+            os.chdir(d)
+            self.assertIsNone(config.repo_owner)
+            os.chdir(cwd)
+
+    def test_repo_branch_from_repo_info(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        config._repo_info = {
+            'branch': 'master',
+        }
+        self.assertEqual(
+            'master', config.repo_branch)
+
+    def test_repo_branch_no_repo_root(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        with temporary_dir() as d:
+            cwd = os.getcwd()
+            os.chdir(d)
+            self.assertIsNone(config.repo_branch)
+            os.chdir(cwd)
+
+    def test_repo_commit_from_repo_info(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        config._repo_info = {
+            'commit': 'abcdef',
+        }
+        self.assertEqual(
+            'abcdef', config.repo_commit)
+
+    def test_repo_commit_no_repo_root(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        with temporary_dir() as d:
+            cwd = os.getcwd()
+            os.chdir(d)
+            self.assertIsNone(config.repo_commit)
+            os.chdir(cwd)
+
+    def test_repo_commit_no_repo_branch(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        with temporary_dir() as d:
+            cwd = os.getcwd()
+            os.chdir(d)
+            os.mkdir(os.path.join(d, '.git'))
+            with open(os.path.join(d, '.git', 'HEAD'), 'w') as f:
+                f.write('abcdef')
+
+            self.assertIsNone(config.repo_commit)
+
+            os.chdir(cwd)
+
+    def test_repo_commit_packed_refs(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        with temporary_dir() as d:
+            cwd = os.getcwd()
+            os.chdir(d)
+            os.system('git init')
+            with open(os.path.join(d, '.git', 'packed-refs'), 'w') as f:
+                f.write('# pack-refs with: peeled fully-peeled sorted\n')
+                f.write('#\n')
+                f.write('8ce67f4519190cd1ec9785105168e21b9599bc27 refs/remotes/origin/master\n')
+
+            self.assertIsNotNone(config.repo_commit)
+
+            os.chdir(cwd)
+
+    def test_use_sentry(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
```

Comment:
i don't completely understand why this test passes? uses the cumlusci root cumulusci.yml?

Reply:
It passes because the mock keychain's `get_service` method returns another mock object without raising an exception.

Reply:
i know you're just fixing up code in place but this indicates something fishy smelling in the code, I think... it's very hard to correlate the setup of the mock to the rest of the code and the observed behavior. I think a comment is probably useful but its also probably a sign we should come back here and do some cleanup.